security/advancedtls/advancedtls.go

```diff
@@ -184,48 +204,89 @@ type ServerOptions struct {
 
 func (o *ClientOptions) config() (*tls.Config, error) {
 	if o.VType == SkipVerification && o.VerifyPeer == nil {
-		return nil, fmt.Errorf(
-			"client needs to provide custom verification mechanism if choose to skip default verification")
+		return nil, fmt.Errorf("client needs to provide custom verification mechanism if choose to skip default verification")
 	}
-	rootCAs := o.RootCACerts
-	if o.VType != SkipVerification && o.RootCACerts == nil && o.GetRootCAs == nil {
-		// Set rootCAs to system default.
-		systemRootCAs, err := x509.SystemCertPool()
-		if err != nil {
-			return nil, err
-		}
-		rootCAs = systemRootCAs
+	// Make sure users didn't specify more than one fields in
+	// RootCertificateOptions and IdentityCertificateOptions.
+	if num := o.RootOptions.nonNilFieldCount(); num > 1 {
+		return nil, fmt.Errorf("at most one field in RootCertificateOptions could be specified")
```

Comment:
It seems this logic is repeated in both client and server options. Can we de-duplicate it?